roofit/roofitcore/src/RooAbsCollection.cxx

```diff
@@ -417,10 +419,9 @@ RooAbsArg *RooAbsCollection::addClone(const RooAbsArg& var, Bool_t silent)
 
   // add a pointer to a clone of this variable to our list (we now own it!)
   auto clone2 = static_cast<RooAbsArg*>(var.Clone());
-  if (clone2) _list.push_back(clone2);
-  if (_allRRV && dynamic_cast<const RooRealVar*>(&var)==0) {
-    _allRRV=kFALSE ;
-  }
+  assert(clone2);
```

Comment:
the switch from `if(clone2)` to `assert(clone2)` is a change in behavior, is it intentional?

Reply:
Yes, the variable should always clone. The `if` should always have been true, and was therefore unnecessary at best. To assert the assumption that a `Clone` in ROOT will not return `nullptr` (not aware that ROOT would do this anywhere), I left an `assert` in place. Makes sense?

Reply:
yes, thanks